src/device.hpp

```diff
@@ -47,6 +47,9 @@ struct cvk_platform;
 
 struct cvk_device : public _cl_device_id,
                     object_magic_header<object_magic::device> {
+    /// Map for storing device pointers to buffer pointers
+    /// Support cl_ext_buffer_device_address
+    std::unordered_map<void*, void*> device_to_buffer_map;
```

Comment:
Those pointers would only be valid for a given context. I think we probably want to keep this state in `cvk_context`. Please use the specific types instead of `void*`.